File: 2-ad-hoc/1091.py
while(True):
    x=1
    x=int(input())
    if x!=0:
        cor_n=[0,0]
        inp=input().split()
        cor_n[0]=int(inp[0]);cor_n[1]=int(inp[1])
        for i in range(x):
            inp=input().split()
            if cor_n[0]==int(inp[0])or cor_n[1]==int(inp[1]): print('divisa')
            else:
                x=int(inp[0])-cor_n[0]
                y=int(inp[1])-cor_n[1]
                prin=''
                if y>0: prin+='N'
                else:prin+='S'
                if x>0: prin+='E'
                else: prin+='O'
                print(prin)
    else:break

File: 2-ad-hoc/1091.cpp
#include<iostream>
#include<stdio.h>
using namespace std;

int main(){
    int x;
    cin>>x;
    while(x!=0){
        int cor_n[2][2];
        cin>>cor_n[0][0]>>cor_n[0][1];
        for(int i=0;i<x;i++){
            cin>>cor_n[1][0]>>cor_n[1][1];
            if(cor_n[0][0]==cor_n[1][0]|cor_n[0][1]==cor_n[1][1]){
                cout<<"divisa"<<endl;
            } else{
                int x=cor_n[1][0]-cor_n[0][0],y=cor_n[1][1]-cor_n[0][1];
                string xy="";
                if(y>0){
                    xy+="N";
                } else{
                    xy+="S";
                }
                if(x>0){
                    xy+="E";
                } else{
                    xy+="O";
                }
                cout<<xy<<endl;
            }
        }
        x=0;
        cin>>x;
    }
}